pkg/core/image.go

```diff
@@ -69,8 +75,58 @@ func (c *imageClient) RelocateImages(options RelocateImagesOptions) error {
 }
 
 func (c *imageClient) DownloadSystem(options DownloadSystemOptions) error {
-	// relocate the manifest mapping no images and without an image manifest
-	return c.relocateManifest(options.Manifest, newIdentityImageMapper(), "", options.Output)
+	return c.downloadSystem(options.Manifest, options.Output)
+}
+
+func (c *imageClient) downloadSystem(manifest string, output string) error {
+	return c.relocateManifest(manifest, newIdentityImageMapper(), "", output)
+}
+
+func (c *imageClient) CreateDistro(options CreateDistroOptions) error {
+	work, err := ioutil.TempDir("", "create_distro_work")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(work)
+
+	fmt.Println("Downloading manifest and kubernetes configuration files...")
+	err = c.downloadSystem(options.Manifest, work)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("\nScanning kubernetes configuration files for images...")
+	imageManifestPath := filepath.Join(work, "image-manifest.yaml")
+	err = c.ListImages(ListImagesOptions{
+		Manifest: filepath.Join(work, "manifest.yaml"),
+		Images:   imageManifestPath,
+		NoCheck:  true, // downloading images will check they are valid
+	})
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("\nDownloading images...")
+	err = c.PullImages(PullImagesOptions{
+		Images: imageManifestPath,
+		Output: work,
+	})
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("\nCreating archive...")
+	err = createArchive(work, options.Output)
+	return nil
+}
+
+func createArchive(dirPath string, archivePath string) error {
+	err := os.MkdirAll(filepath.Dir(archivePath), outputDirPermissions)
```

Comment:
if the achivePath is a directory, should we err or provide a default filename? I mistakenly ran the command passing a directory and was confused as to why it didn't create anything.

Reply:
A default filename seems sensible, but I'll avoid "riff" in the name because of the new flexible CLI naming. Probably use "distro.tgz".